dali/pipeline/workspace/workspace.h

```diff
@@ -203,6 +203,28 @@ class WorkspaceBase : public ArgumentWorkspace {
                                      StorageDevice::GPU);
   }
 
+
+  /**
+   * @brief Returns true if this workspace has CUDA stream available
+   */
+  virtual bool has_stream() const = 0;
+
+
+  /**
+   * @brief Returns the CUDA stream that this work is to be done in.
+   */
+  cudaStream_t stream() const {
```

Comment:
You still will get exception in your face if you call this in CPU, but I guess it's better if you can compile it.